### PR TITLE
Add addons to AVS 2020-07-17-preview

### PR DESCRIPTION
**Pull request by leedouglas**
, moving it to 2020-07-17-preview API from PR repo

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ChenTanyi** [comment]
@leedouglas Is it from the PR in private repo? If so, please link them together.

And also please fix the CI error (except breaking change if the breaking change is expected)

**Comment by leedouglas** [comment]
@ChenTanyi 
Breaking change is expected.   We are working on CI errors.

**Comment by ctaggart** [comment]
@leedouglas , I fixed the systemData errors. I was able to get the errors by running the [validator](https://github.com/Azure/azure-openapi-validator) like so:
```
autorest --validation --azure-validator --use=@microsoft.azure/classic-openapi-validator@latest --use=@microsoft.azure/openapi-validator@latest --tag=package-2020-07-17-preview specification\vmware\resource-manager\readme.md
```

@ChenTanyi, the addon changes were reviewed in https://github.com/Azure/azure-rest-api-specs-pr/pull/1593, but were mistakenly left out of https://github.com/Azure/azure-rest-api-specs/pull/10866. The API endpoints are in use by the Portal UI in production. The mistake was caught by the S360 validation. This will resolve the liveness validation errors. Please review. I'll also add this to be in a breaking change review next Monday.

**Comment by ChenTanyi** [comment]
Got it. I saw it also added some new API, which would need ARM review it.

Meanwhile, could you please fix the SDK error? It seems you define two properties as same name in "PrivateCloudProperties", but one is string and the other (defined in parent schema "PrivateCloudUpdateProperties") is enum.
```
cmderr	[generate.py] ERROR (PreCheck/PropertyRedeclaration): Schema 'PrivateCloudProperties' has a property 'vcenterPassword' that is conflicting with a property in the parent schema 'PrivateCloudUpdateProperties' differs more than just description : [enum => '<removed>',x-ms-enum => '<removed>',x-ms-secret => 'true']
cmderr	[generate.py] ERROR (PreCheck/PropertyRedeclaration): Schema 'PrivateCloudProperties' has a property 'nsxtPassword' that is conflicting with a property in the parent schema 'PrivateCloudUpdateProperties' differs more than just description : [enum => '<removed>',x-ms-enum => '<removed>',x-ms-secret => 'true']
```

**Comment by ctaggart** [comment]
Per internal ARM wiki guidance, [System Metadata](https://armwiki.azurewebsites.net/rpaas/systemdata.html) should go in a new API version, so I've suppressed the error. It will be in the next API, the 2021-01-01-preview API.

@ChenTanyi, the SDK errors related to `PrivateCloudProperties` are unfortunate. It is a mistake that will be resolved in the next preview API version ([internal work item 9034355](https://msazure.visualstudio.com/One/_workitems/edit/9034355/)). This pull request does not and will not address it. This pull request is just about adding the missing addons endpoints.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json [hunk 1]
```diff
@@ -3957,6 +4165,110 @@
...
+    "AddonProperties": {
+      "description": "The properties of an addon that may be updated",
+      "allOf": [
+        {
+          "$ref": "#/definitions/AddonSrmProperties"
+        }
+      ],
+      "properties": {
+        "systemData": {
+          "description": "required by CI process",
```
Comment (jorgecotillo) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json]: Is this expected? "properties" included twice?

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json [hunk 2]
```diff
@@ -3162,6 +3170,206 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "successful operation",
+            "schema": {
+              "$ref": "#/definitions/AddonList"
```
Comment (jorgecotillo) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json:3202]: you are exposing new api functionality, this should be modeled as a new api-version, see the bot's recommendation as well (NewApiVersionRequired)

**Comment by ChenTanyi** [comment]
> @ChenTanyi, the SDK errors related to `PrivateCloudProperties` are unfortunate. It is a mistake that will be resolved in the next preview API version ([internal work item 9034355](https://msazure.visualstudio.com/One/_workitems/edit/9034355/)). This pull request does not and will not address it. This pull request is just about adding the missing addons endpoints.

@ctaggart If you don't need Java SDK for this API version, then it will be OK.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json (hunk 1 above)
Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json]: This was already removed. `systemData` will be in the next API version, not this one. See commit 3c0ca92.

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json (hunk 2 above)
Comment (ctaggart) [reply specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json:3202]: It was in this version in the pull request for the private repo. It was left out by mistake. It is already being used in the portal UI in production. Please see my [earlier comments.](https://github.com/Azure/azure-rest-api-specs/pull/12236#issuecomment-751894356). The easiest solution is to add the changes back in to this API version.

**Comment by ctaggart** [comment]
> If you don't need Java SDK for this API version, then it will be OK.

@ChenTanyi, cool, we don't need the Java SDK for this API version.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ctaggart** [comment]
@ChenTanyi , the breaking change has been approved. It was got ARM signoff. Please merge.

**Pull request merged**